compiler/rustc_query_system/src/dep_graph/dep_kind.rs

```diff
@@ -0,0 +1,169 @@
+}
+
+#[allow(non_upper_case_globals)]
+pub mod dep_kind {
+    use super::*;
+
+    // We use this for most things when incr. comp. is turned off.
+    pub const Null: DepKindStruct =
+        DepKindStruct { has_params: false, is_anon: false, is_eval_always: false };
+
+    pub const TraitSelect: DepKindStruct =
+        DepKindStruct { has_params: false, is_anon: true, is_eval_always: false };
+
+    pub const CompileCodegenUnit: DepKindStruct =
+        DepKindStruct { has_params: true, is_anon: false, is_eval_always: false };
+
+    pub const CompileMonoItem: DepKindStruct =
+        DepKindStruct { has_params: true, is_anon: false, is_eval_always: false };
+
+    macro_rules! define_query_dep_kinds {
+        ($(
+            [$($attrs:tt)*]
+            $variant:ident $(( $tuple_arg_ty:ty $(,)? ))*
+        ,)*) => (
+            $(pub const $variant: DepKindStruct = {
+                const has_params: bool = $({ erase!($tuple_arg_ty); true } |)* false;
+                const is_anon: bool = contains_anon_attr!($($attrs)*);
+                const is_eval_always: bool = contains_eval_always_attr!($($attrs)*);
+
+                DepKindStruct {
+                    has_params,
+                    is_anon,
+                    is_eval_always,
+                }
+            };)*
+        );
+    }
+
+    rustc_dep_node_append!([define_query_dep_kinds!][]);
+}
+
+macro_rules! define_dep_nodes {
+    (<$tcx:tt>
+    $(
+        [$($attrs:tt)*]
+        $variant:ident $(( $tuple_arg_ty:ty $(,)? ))*
+      ,)*
+    ) => (
+        #[macro_export]
+        macro_rules! make_dep_kind_array {
+            ($mod:ident) => {[ $(($mod::$variant),)* ]};
+        }
+
+        static DEP_KINDS: &[DepKindStruct] = &make_dep_kind_array!(dep_kind);
+
+        /// This enum serves as an index into the `DEP_KINDS` array.
+        #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, Encodable, Decodable)]
+        #[allow(non_camel_case_types)]
+        pub enum DepKind {
+            $($variant),*
+        }
+
+        pub fn dep_kind_from_label_string(label: &str) -> Result<DepKind, ()> {
+            match label {
+                $(stringify!($variant) => Ok(DepKind::$variant),)*
+                _ => Err(()),
+            }
+        }
+
+        /// Contains variant => str representations for constructing
+        /// DepNode groups for tests.
+        #[allow(dead_code, non_upper_case_globals)]
+        pub mod label_strs {
+           $(
+                pub const $variant: &str = stringify!($variant);
+            )*
+        }
+    );
+}
+
+rustc_dep_node_append!([define_dep_nodes!][ <'tcx>
+    // We use this for most things when incr. comp. is turned off.
+    [] Null,
+
+    [anon] TraitSelect,
+
+    // WARNING: if `Symbol` is changed, make sure you update `make_compile_codegen_unit` below.
+    [] CompileCodegenUnit(Symbol),
+
+    // WARNING: if `MonoItem` is changed, make sure you update `make_compile_mono_item` below.
```

Comment:
These comments need to be updated as `make_compile_*` are no longer defined in the same file.